apps/test-netzer-encoder/src/Netzer.cpp

```diff
@@ -0,0 +1,57 @@
+#include "Netzer.h"
+
+using namespace Encoder;
+Netzer::Netzer(PinName mosi, PinName miso, PinName sclk, 
+									callback_ptr callback) 
+									: m_spi(mosi, miso, sclk), m_callback(callback){
+	m_spi.format(16,3);
+	m_spi.frequency(DEFAULT_FREQUENCY_HZ);
+	m_spi.set_dma_usage(DMA_USAGE_ALWAYS);
+}
+
+bool Netzer::spi_read(){
+
+	ThisThread::sleep_for(3ms);		// Delay
+
+	// return status
+	int status = m_spi.transfer(tx_buffer, WORDS, rx_buffer, WORDS, event_callback_t(this, &Netzer::Netzer::spi_callback_debug));
+	
+	// status = 0 => SPI transfer started
+  // status = -1 => SPI peripheral is busy
+	return (status == 0);
+}
+
+uint16_t Netzer::get_raw_data(){
+	return m_raw_data;
+}
+
+void Netzer::spi_callback_debug(int events){
+	// upper lower?
+	const uint8_t crcPolynomial = 0x03;
```

Comment:
In his last commit Alex said that the CRC value was wrong XD

Reply:
oh lol gotcha, I should probably figure out what CRC value is hehe